ion/marshal.go

```diff
@@ -352,6 +359,30 @@ func (m *Encoder) encodeDecimal(v reflect.Value) error {
 	return m.w.WriteDecimal(d)
 }
 
+func (m *Encoder) encodeWithAnnotation(v reflect.Value) error {
+	original := v
+	fields := fieldsFor(v.Type())
+	for _, field := range fields {
+		if field.annotations {
+			annotations, err := findSubvalue(original, &field)
+			if err != nil {
+				return err
+			}
+			if annotations.Kind() != reflect.Slice {
+				return fmt.Errorf("ion: '%v' is provided for annotations, it must be []string", annotations.Kind())
```

Comment:
```suggestion
				return fmt.Errorf("ion: '%v' is provided for annotations, it must be of type []string", annotations.Kind())
```

Reply:
Oops, I did not see the suggested commit, I included the change in [this commit](039acd2f9ab1ad204e61ef1f78a4c2186b377d49).